airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/initialsync/MssqlInitialLoadRecordIterator.java

```diff
@@ -0,0 +1,166 @@
+  private final JdbcCompatibleSourceOperations<JDBCType> sourceOperations;
+  private final List<String> columnNames;
+  private final AirbyteStreamNameNamespacePair pair;
+  private final MssqlInitialLoadStateManager initialLoadStateManager;
+  private final long chunkSize;
+  private final OrderedColumnInfo ocInfo;
+  private final boolean isCompositeKeyLoad;
+
+  MssqlInitialLoadRecordIterator(
+                                 final JdbcDatabase database,
+                                 final JdbcCompatibleSourceOperations<JDBCType> sourceOperations,
+                                 final String quoteString,
+                                 final MssqlInitialLoadStateManager initialLoadStateManager,
+                                 final List<String> columnNames,
+                                 final AirbyteStreamNameNamespacePair pair,
+                                 final long chunkSize,
+                                 final boolean isCompositeKeyLoad) {
+    this.database = database;
+    this.sourceOperations = sourceOperations;
+    this.quoteString = quoteString;
+    this.initialLoadStateManager = initialLoadStateManager;
+    this.columnNames = columnNames;
+    this.pair = pair;
+    this.chunkSize = chunkSize;
+    this.ocInfo = initialLoadStateManager.getOrderedColumnInfo(pair);
+    this.isCompositeKeyLoad = isCompositeKeyLoad;
+  }
+
+  @CheckForNull
+  @Override
+  protected JsonNode computeNext() {
+    if (shouldBuildNextSubquery()) {
+      try {
+        // We will only issue one query for a composite key load. If we have already processed all the data
+        // associated with this
+        // query, we should indicate that we are done processing for the given stream.
+        if (isCompositeKeyLoad && numSubqueries >= 1) {
+          return endOfData();
+        }
+        // Previous stream (and connection) must be manually closed in this iterator.
+        if (currentIterator != null) {
+          currentIterator.close();
+        }
+
+        LOGGER.info("Subquery number : {}", numSubqueries);
+        final Stream<JsonNode> stream = database.unsafeQuery(
+            this::getOcPreparedStatement, sourceOperations::rowToJson);
+        currentIterator = AutoCloseableIterators.fromStream(stream, pair);
+        numSubqueries++;
+        // If the current subquery has no records associated with it, the entire stream has been read.
+        if (!currentIterator.hasNext()) {
+          return endOfData();
+        }
+      } catch (final Exception e) {
+        throw new RuntimeException(e);
+      }
+    }
+    return currentIterator.next();
+  }
+
+  private boolean shouldBuildNextSubquery() {
+    // The next sub-query should be built if (i) it is the first subquery in the sequence. (ii) the
+    // previous subquery has finished.
+    return (currentIterator == null || !currentIterator.hasNext());
+  }
+
+  private PreparedStatement getOcPreparedStatement(final Connection connection) {
```

Comment:
what does OC stand for?

Reply:
OrderedColumn (the differentiate from PK)